### ISSUE TITLE
WBXML_NAMESPACE_SEPARATOR ':' incompatible with Expat >=2.4.5 security fixes

### ISSUE THREAD

**Issue by hartwork** [libwbxml/libwbxml#76]
Hi libwbxml,

it was [brought to my attention](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1006337) that libwbxml and Expat >=2.4.5 are not compatible. The incompatibility has two halves:
- libwbxml is using `:` (colon) for a namespace separator at https://github.com/libwbxml/libwbxml/blob/e0d845bf4abf6f02a50737be9db9c60527c530ed/src/wbxml_internals.h#L159
- Expat disallows use of namespace separator characters in `xmlns` attribute values for security since 2.4.5 and PR https://github.com/libexpat/libexpat/pull/561.

The [API docs of `XML_ParserCreateNS`](https://libexpat.github.io/doc/api/latest/#XML_ParserCreateNS) say:

> This means that you should pick a character for sep that can't be part of an URI.

Colon can appear in a URI (RFC 3986), so a colon cannot be used as a namespace separator or there will be false positives. Use of `" "` (space) or `"\n"` (line feed) would work.

Would you be up for picking a different character for a namespace separator?

Thanks and best, Sebastian

**Comment by SlavekB** [comment]
When I try to use another separator than `':'` (I tried `'|'`, `'\n'`, `' '`, `'\'`), there is a tests failure because call `wbxml2xml` ends on `Segmentation Fault`:
```
The following tests FAILED:
         22 - tools_document_ddf_2 (Failed)
         91 - tools_document_syncml_3 (Failed)
         92 - tools_document_syncml_4 (Failed)
```
This will require further effort than just a simple change of the separator.

Cheers,
Slávek

**Comment by hartwork** [comment]
Hi @SlavekB thanks for bringing that to light. I have zero prior knowledge of libwbxml myself, but I'm happy to team up on this issue.

**Comment by SlavekB** [comment]
Although the Segmentation Fault can be prevented by a simple patch…
```
--- wbxml2-0.11.6.orig/src/wbxml_parser.c
+++ wbxml2-0.11.6/src/wbxml_parser.c
@@ -2479,7 +2479,8 @@ static WBXMLError decode_opaque_content(
     case WBXML_LANG_SYNCML_SYNCML11:
     case WBXML_LANG_SYNCML_SYNCML12:
         /* NextNonce */
-        if ((parser->current_tag->wbxmlCodePage == 0x01) &&
+        if ((parser->current_tag) &&
+            (parser->current_tag->wbxmlCodePage == 0x01) &&
             (parser->current_tag->wbxmlToken == 0x10))
         {
             /* Decode base64 value */
```
…it only solves the consequence, but not the cause – generated wbxml is not correct.

**Comment by hartwork** [comment]
Closing as fixed due to #78 being merged by now. Please re-open if you see need to.
